code/network/multi_obj.cpp

```diff
@@ -496,25 +508,30 @@ int multi_ship_record_find_frame(int client_frame, int time_elapsed)
 	for (int i = Oo_info.cur_frame_index - 2; i > -1; i--) {
 
 		// Check to see if the client's timestamp matches the recorded frames.
-		if ((Oo_info.timestamps[i] <= target_timestamp) && (Oo_info.timestamps[i + 1] > target_timestamp)) {
+		if ((Oo_info.timestamps[i] <= target_timestamp) && (Oo_info.timestamps[i + 1] >= target_timestamp)) {		
```

Comment:
This looks like it could benefit from a timestamp ordering check.  I'll add that to the `timestamp_since` PR.

Reply:
Or a new PR.  See #4324.